libs/platform/ebpf_object.c

```diff
@@ -172,7 +188,9 @@ ebpf_object_release_reference(ebpf_object_t* object)
     new_ref_count = ebpf_interlocked_decrement_int32(&object->reference_count);
 
     if (new_ref_count == 0) {
+        ebpf_lock_state_t state = ebpf_lock_lock(&_ebpf_object_tracking_list_lock);
         _ebpf_object_tracking_list_remove(object);
```

Comment:
suggestion: We can also call `_ebpf_object_release_reference_under_lock()` here after acquiring lock?

Reply:
if we did that literally here, it would double decrement.  The point currently is to only take the lock here when the refcount hits 0 so we want the decrement outside the lock in this case.  That's why I had to have 2 separate functions for now.